crates/sui-types/src/test_checkpoint_data_builder.rs

```diff
@@ -626,10 +629,39 @@ impl TestCheckpointDataBuilder {
             &committee,
         );
         self.checkpoint_builder.checkpoint += 1;
-        CheckpointData {
-            checkpoint_summary: checkpoint_cert,
-            checkpoint_contents: contents,
-            transactions,
+
+        // Build the object set and convert transactions to ExecutedTransaction
+        let mut object_set = ObjectSet::default();
+        let executed_transactions = transactions
+            .into_iter()
+            .map(|tx| {
+                // Insert all input and output objects into the object set
+                for o in tx
+                    .input_objects
+                    .into_iter()
+                    .chain(tx.output_objects.into_iter())
```

Comment:
nit: is the `into_iter` required here? I think you can pass anything that implements `IntoIterator` in.